tests/unit/CollectionResolverTest.php

```diff
@@ -0,0 +1,252 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection;
+
+use Mockery as m;
+use phpDocumentor\Reflection\Types\Array_;
+use phpDocumentor\Reflection\Types\Collection;
+use phpDocumentor\Reflection\Types\Compound;
+use phpDocumentor\Reflection\Types\Context;
+use phpDocumentor\Reflection\Types\Iterable_;
+use phpDocumentor\Reflection\Types\Nullable;
+use phpDocumentor\Reflection\Types\Object_;
+use phpDocumentor\Reflection\Types\Boolean;
+use Mockery\MockInterface;
+use phpDocumentor\Reflection\Types\String_;
+
+/**
+ * @coversDefaultClass phpDocumentor\Reflection\TypeResolver
+ */
+class CollectionResolverTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     *
+     * @covers ::__construct
+     * @covers ::resolve
+     * @covers ::<private>
```

Comment:
When you move this `::<private>` to the class level you don't have to repeat it every time.